### PR TITLE
`Development`: Add documentation about notifications

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Description

Adds description about notifications

### Steps for Testing

1. Look at the documentation about notifications under the communication tab here: https://artemis-platform--5351.org.readthedocs.build/en/5351/

### Review Progress

- [x] Review 1
- [x] Review 2

### CODE REVIEW

**Review by Marvin-Vitcu — APPROVED** [review]
Looks good. Well explained.

One thing that bothers me with the documentation in general:
The headlines have the same distance to the text above as they have to the text below. This is a bit confusing. It would be much better if the headlines were closer to the text below (the text that the headline actually belongs to).

File: docs/user/communication.rst [hunk 1]
```diff
@@ -142,9 +142,48 @@ Announcements can be created in the course communication overview by selecting t
...
+Notifications
+-------------
+
+Artemis supports customizable web and email notifications. Users can enable and disable different notification types.
+
+Web Notifications
+^^^^^^^^^^^^^^^^^
+
+The web notifications can be found on the top right of the page by clicking on the bell icon.
+A red number shows the number of new messages.
```
Comment (Strohgelaender) [inline docs/user/communication.rst:154]: number twice
```suggestion
A red number shows the amount of new messages.
```

File: docs/user/communication.rst [hunk 2]
```diff
@@ -142,9 +142,48 @@ Announcements can be created in the course communication overview by selecting t
...
+Artemis can also send out emails for certain notification types.
+Additionally, Artemis can send out a weekly summary at Friday 5pm.
+
+|notification-email|
+
+Settings
+^^^^^^^^
+
+The user can change their preference for different types of notifications and decide if they want to receive emails, web notifications or no notification.
+These settings can be found after clicking on the web notifications. The gear on the top left of the sidebar then leads to the settings.
```
Comment (Strohgelaender) [inline docs/user/communication.rst:172]:
```suggestion
These settings can be found after opening the web notifications. The gear on the top left of the sidebar then leads to the settings.
```

**Review by Strohgelaender — COMMENTED** [review]
The screenshots (at least for the e-mail) should not contain real notifications and show made-up testvalues instead.

File: docs/user/communication.rst (hunk 1 above)
Comment (b-fein) [reply docs/user/communication.rst:154]: I think number is actually correct here, as amount is used for uncountable elements (‘Anzahl’ vs ‘Menge’ in German).

**Review by manuelmanso — COMMENTED** [review]
I like your additions. But like @Strohgelaender mentioned I think you should be using the [personas ](https://confluence.ase.in.tum.de/pages/viewpage.action?spaceKey=ArTEMiS&title=Random+Personas)we discussed in the developer meeting for your screenshots.

**Comment by Dominik-Weinzierl** [comment]
You don't need to create new users, just simply inspect the html code and change the names :+1: (easiest solution)

**Review by MetinLamby — COMMENTED** [review]
under the heading "settings" -> sentence "These settings can be found after clicking on the web notifications"

When saying "after clicking on the web notifications" maybe you can refer to the above "web notifications" header so that the student clearly understands what is meant by "clicking on the web notifications"

**Review by nadinesamyy — DISMISSED** [review]
Easily understandable and clear, maybe in the web notifications section you could add that when the user clicks on the bell a menu drop down list appears

**Review by jschmautzer — APPROVED** [review]
understandable :)

only one small thing: Under the section "Settings", it should be "The **users** can change their ..." instead of "The **user** can change their ..."

**Comment by JohannesStoehr** [comment]
> understandable :)
> 
> only one small thing: Under the section "Settings", it should be "The **users** can change their ..." instead of "The **user** can change their ..."

That formulation is intended. You can also use they for singular instead of his/her to keep the text gender neutral

**Review by luisa-we — APPROVED** [review]
Very simple and understandable explanation.

**Review by jbllhff — DISMISSED** [review]
Looks good, thanks!

**Review by rhubinger — APPROVED** [review]
Read the page, looks good.

**Review by jschmautzer — APPROVED** [review]
looks good to me

File: docs/user/notifications.rst [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+.. contents:: Content of this document
+    :local:
+    :depth: 2
+
+Artemis supports customizable web and email notifications. Users can enable and disable different notification types.
+
+Web Notifications
+^^^^^^^^^^^^^^^^^
+
+The web notifications can be found on the top right of the page by clicking on the bell icon.
```
Comment (babyygemperor) [inline docs/user/notifications.rst:15]: Click is a transitive verb.
By clicking the bell icon

File: docs/user/notifications.rst [hunk 4]
```diff
@@ -0,0 +1,45 @@
+.. _notifications:
+
+Notifications
+=============
+
+.. contents:: Content of this document
+    :local:
+    :depth: 2
+
+Artemis supports customizable web and email notifications. Users can enable and disable different notification types.
```
Comment (babyygemperor) [inline docs/user/notifications.rst:10]: Types of notification sounds more natural. Just like in [line 33](https://github.com/ls1intum/Artemis/blob/936154fc747be74638e2644952fdcaf0e48f93a9/docs/user/notifications.rst?plain=1?#L33)

File: docs/user/notifications.rst [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+A red indicator shows the number of new messages.
+
+|notification-top-bar|
+
+|notification-side-bar|
+
+Email Notifications
+^^^^^^^^^^^^^^^^^^^
+
+Artemis can also send out emails for certain notification types.
```
Comment (babyygemperor) [inline docs/user/notifications.rst:25]: Same as the one above

**Pull request merged**